compiler/incremental-compilation-impl/src/org/jetbrains/kotlin/incremental/IncrementalCachesManager.kt

```diff
@@ -51,33 +50,28 @@ abstract class IncrementalCachesManager<PlatformCache : AbstractIncrementalCache
         LookupStorage(lookupCacheDir, pathConverter, storeFullFqNamesInLookupCache, trackChangesInLookupCache).apply { registerCache() }
     abstract val platformCache: PlatformCache
 
+    @Suppress("UnstableApiUsage")
     @Synchronized
-    fun close(flush: Boolean = false): Boolean {
-        if (isClosed) {
-            return isSuccessfulyClosed
-        }
-        isSuccessfulyClosed = true
-        for (cache in caches) {
-            if (flush) {
-                try {
-                    cache.flush(false)
-                } catch (e: Throwable) {
-                    isSuccessfulyClosed = false
-                    reporter.info { "Exception when flushing cache ${cache.javaClass}: $e" }
-                }
-            }
+    override fun close() {
+        check(!isClosed) { "This cache storage has already been closed" }
 
-            try {
-                cache.close()
-            } catch (e: Throwable) {
-                isSuccessfulyClosed = false
-                reporter.info { "Exception when closing cache ${cache.javaClass}: $e" }
-            }
+        val closer = Closer.create()
+        caches.forEach {
+            closer.register(CacheCloser(it))
         }
+        closer.close()
 
         isClosed = true
-        return isSuccessfulyClosed
     }
+
+    private class CacheCloser(private val cache: BasicMapsOwner) : Closeable {
+
+        override fun close() {
+            cache.flush(memoryCachesOnly = false)
```

Comment:
Do we want to always flush?

Reply:
I searched for existing usages of `IncrementalCachesManager.close()` and saw that we always called `IncrementalCachesManager.close(flush = true)`, so this is preserving the existing behavior (btw, `memoryCachesOnly = false` is also preserving the existing behavior). We can re-introduce the flushing flag later if there is a need not to flush it. // cc @ausatiy

Reply:
Amazingly, I just found out that the IC instability that we were seeing only in KGP 1.7.10 was caused by *not* flushing the caches when closing them (i.e., https://youtrack.jetbrains.com/issue/KT-53168, which was fixed by https://github.com/jetbrains/kotlin/commit/5cae57cf28675720710273b914efa08d7e5f5bbb).

Reply:
Thanks for sharing this! Yeah, with this in mind it is a safe thing to always flush.